improver/utilities/temporal.py

```diff
@@ -254,3 +266,217 @@ def extract_nearest_time_point(cube, dt, time_name="time", allowed_dt_difference
     constr = iris.Constraint(coord_values={time_name: nearest_dt})
     cube = cube.extract(constr)
     return cube
+
+
+class TimezoneExtraction(PostProcessingPlugin):
+    """Plugin to extract local time offsets"""
+
+    def __init__(self):
+        self.time_coord_standards = TIME_COORDS["time"]
+        self.time_points = None
+        self.time_bounds = None
+        self.time_units = cf_units.Unit(
+            self.time_coord_standards.units,
+            calendar=self.time_coord_standards.calendar,
+        )
+        self.timezone_cube = None
+
+        self.output_data = None
+
+    def create_output_cube(self, cube, local_time):
+        """
+        Constructs the output cube
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube of data to extract timezone-offsets from. Must contain a time
+                coord spanning all the timezones.
+            local_time (datetime.datetime):
+                The "local" time of the output cube as %Y%m%dT%H%MZ. This will form a
+                scalar "time_in_local_timezone" coord on the output cube, while the
+                "time" coord will be auxillary to the spatial coords and will show the
+                UTC time that matches the local_time at each point.
+
+        """
+        template_cube = cube.slices_over("time").next().copy()
+        template_cube.remove_coord("time")
+        template_cube.remove_coord("forecast_period")
+        output_cube = create_new_diagnostic_cube(
+            template_cube.name(),
+            template_cube.units,
+            template_cube,
+            generate_mandatory_attributes([template_cube]),
+            optional_attributes=template_cube.attributes,
+            data=self.output_data,
+        )
+
+        # Create a local time coordinate to help with plotting data.
+        local_time_coord_standards = TIME_COORDS["time_in_local_timezone"]
+        local_time_units = cf_units.Unit(
+            local_time_coord_standards.units,
+            calendar=local_time_coord_standards.calendar,
+        )
+        timezone_points = np.array(
+            np.round(local_time_units.date2num(local_time)),
+            dtype=local_time_coord_standards.dtype,
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                timezone_points,
+                long_name="time_in_local_timezone",
+                units=local_time_units,
+            )
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                self.time_points,
+                bounds=self.time_bounds,
+                standard_name="time",
+                units=self.time_units,
+            ),
+            (0, 1),
+        )
+        return output_cube
+
+    def fill_timezones(self, input_cube):
```

Comment:
This is not unit tested explicitly, which is fine as it is the bulk of the process method, but it should probably be made a private method for that reason.

Reply:
Changed this one too.